pkg/storegateway/bucket.go

```diff
@@ -880,109 +878,18 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 	if s.debugLogging {
```

Comment:
I think the following block of code can be moved before the call to `s.mtx.RLock()`:

```
	spanLogger := spanlogger.FromContext(srv.Context(), s.logger)
	level.Debug(spanLogger).Log(
		"msg", "BucketStore.Series",
		"request min time", time.UnixMilli(req.MinTime).UTC().Format(time.RFC3339Nano),
		"request max time", time.UnixMilli(req.MaxTime).UTC().Format(time.RFC3339Nano),
		"request matchers", storepb.PromMatchersToString(matchers...),
		"request shard selector", maybeNilShard(shardSelector).LabelValue(),
	)
```